networkx/generators/tests/test_time_series.py

```diff
@@ -0,0 +1,75 @@
+"""Unit tests for the :mod:`networkx.generators.time_series` module."""
+import itertools
+
+import networkx as nx
+
+
+def test_visibility_graph__empty_series__empty_graph():
+    null_graph = nx.visibility_graph([])
+    assert nx.is_empty(null_graph)
+
+
+def test_visibility_graph__single_value_ts__single_node_graph():
+    node_graph = nx.visibility_graph([10])
+    assert node_graph.number_of_nodes() == 1
+    assert node_graph.number_of_edges() == 0
+
+
+def test_visibility_graph__two_values_ts__single_edge_graph():
+    edge_graph = nx.visibility_graph([10, 20])
+    assert list(edge_graph.edges) == [(0, 1)]
+
+
+def test_visibility_graph__convex_series__complete_graph():
+    # arrange
+    series = [i**2 for i in range(10)]  # no obstructions
+    expected_series_length = len(series)
+
+    # act
+    actual_graph = nx.visibility_graph(series)
+
+    # assert
+    assert actual_graph.number_of_nodes() == expected_series_length
+    assert actual_graph.number_of_edges() == 45
+    assert nx.is_isomorphic(actual_graph, nx.complete_graph(expected_series_length))
+
+
+def test_visibility_graph__concave_series__path_graph():
+    # arrange
+    series = [-(i**2) for i in range(10)]  # living in 1D flatland
+    expected_node_count = len(series)
+
+    # act
```

Comment:
Take out the comments "arrange", "act", "assert" because the variable names tell you what is going on.